### PR TITLE
SUP-27588-replaced-entry-drop-folder-file: Change query for replacing entry drop folder file

### PR DESCRIPTION
**Pull request by OrelAri**
Drop Folder File been created for replaced entryId. 
CodeFix for checking if the entry is temporary if so taking the replaced entryID for the query.

### CODE REVIEW

File: plugins/drop_folder/lib/kDropFolderEventsConsumer.php [hunk 1]
```diff
@@ -266,6 +266,10 @@ private function onEntryStatusChanged( $entry )
 	{
 		// Handle only files that are still in the PROCESSING state, which were left
 		// in this state due to AUTO_DELETE_WHEN_ENTRY_IS_READY delete policy.
+        if ($entry->getIsTemporary())
+        {
+            $entry = $entry->getReplacedEntryId();
+        }
```
Comment (amiras89) [inline plugins/drop_folder/lib/kDropFolderEventsConsumer.php:272]: @OrelAri I think this is fragile + you place entryId value into what was an $entry object
Better to

Please consider using the fix I suggested when we investigated it:
if ($entry->getIsTemporary() && $entry->getReplacedEntryId())
        {
            $entryId = $entry->getReplacedEntryId();
        }
        else
        {
            $entryId = $entry->getId();
        }

File: plugins/drop_folder/lib/kDropFolderEventsConsumer.php [hunk 2]
```diff
@@ -266,6 +266,10 @@ private function onEntryStatusChanged( $entry )
...
 		$dropFolderFiles = DropFolderFilePeer::retrieveByEntryIdPartnerIdAndStatuses($entry->getId(), $entry->getPartnerId(), array(DropFolderFileStatus::PROCESSING));
```
Comment (amiras89) [inline plugins/drop_folder/lib/kDropFolderEventsConsumer.php:273]: Need to modify $entry->getId to '$entryId' (based on the above fix suggestion)
$dropFolderFiles = DropFolderFilePeer::retrieveByEntryIdPartnerIdAndStatuses($entryId, $entry->getPartnerId(), array(DropFolderFileStatus::PROCESSING));

+ need to further inspect the rest of this function to see if other alignments should be made

File: plugins/drop_folder/lib/kDropFolderEventsConsumer.php [hunk 3]
```diff
@@ -266,7 +266,17 @@ private function onEntryStatusChanged( $entry )
 	{
 		// Handle only files that are still in the PROCESSING state, which were left
 		// in this state due to AUTO_DELETE_WHEN_ENTRY_IS_READY delete policy.
-		$dropFolderFiles = DropFolderFilePeer::retrieveByEntryIdPartnerIdAndStatuses($entry->getId(), $entry->getPartnerId(), array(DropFolderFileStatus::PROCESSING));
+        if ($entry->getIsTemporary() && $entry->getReplacedEntryId())
+        {
+            $entryId = $entry->getReplacedEntryId();
+            KalturaLog::debug('Found replacedEntryId:'.$entryId);
```
Comment (MosheMaorKaltura) [inline plugins/drop_folder/lib/kDropFolderEventsConsumer.php:272]: Add more informative message

Comment (MosheMaorKaltura) [reply plugins/drop_folder/lib/kDropFolderEventsConsumer.php:272]: space before and after .

File: plugins/drop_folder/lib/kDropFolderEventsConsumer.php [hunk 4]
```diff
@@ -266,7 +266,17 @@ private function onEntryStatusChanged( $entry )
 	{
 		// Handle only files that are still in the PROCESSING state, which were left
 		// in this state due to AUTO_DELETE_WHEN_ENTRY_IS_READY delete policy.
-		$dropFolderFiles = DropFolderFilePeer::retrieveByEntryIdPartnerIdAndStatuses($entry->getId(), $entry->getPartnerId(), array(DropFolderFileStatus::PROCESSING));
+        if ($entry->getIsTemporary() && $entry->getReplacedEntryId())
+        {
```
Comment (MosheMaorKaltura) [inline plugins/drop_folder/lib/kDropFolderEventsConsumer.php:270]: Use only tabs

**Comment by MosheMaorKaltura** [comment]
reviewed

**Pull request merged**